tests/test_mcore_utilities.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE: Copied from https://github.com/NVIDIA/Megatron-LM/blob/main/tests/unit_tests/test_utilities.py
+# to avoid having to rely on megatron repo for tests when all we need is this file
+
+import os
+from datetime import timedelta
+
+import megatron.core.parallel_state as ps
+import torch
+from torch._C._distributed_c10d import PrefixStore
+from torch.distributed import rendezvous
+from torch.distributed.distributed_c10d import _store_based_barrier
+
+
+class TestModel(torch.nn.Module):
+    def __init__(self, input_dim: int, output_dim: int, num_layers: int, bias: bool):
+        super().__init__()
+        self.layers = torch.nn.ModuleList([torch.nn.Linear(input_dim, output_dim, bias) for _ in range(num_layers)])
+
+
+class Utils:
+
+    world_size = torch.cuda.device_count()
+    rank = int(os.environ["LOCAL_RANK"])
+    inited = False
+    store = None
+
+    @staticmethod
+    def initialize_distributed():
+        if not torch.distributed.is_initialized() and Utils.rank >= 0:
+            print(f"Initializing torch.distributed with rank: {Utils.rank}, world_size: {Utils.world_size}")
+            torch.cuda.set_device(Utils.rank % torch.cuda.device_count())
+            init_method = "tcp://"
+            master_ip = os.getenv("MASTER_ADDR", "localhost")
+            master_port = os.getenv("MASTER_PORT", "6000")
+            init_method += master_ip + ":" + master_port
+            rendezvous_iterator = rendezvous(init_method, Utils.rank, Utils.world_size, timeout=timedelta(minutes=1))
+            store, rank, world_size = next(rendezvous_iterator)
```

Comment:
nit: rank, world_size seems unused

Reply:
Addressed in f16b4cc4b793107d4e23e639cfea29448fd19236